File: gems/decomplex/examples/ruby/co-update.rb
# frozen_string_literal: true

def stable_one(node)
  node.storage = :heap
  node.provenance = :heap
end

def stable_two(node)
  node.storage = :heap
  node.provenance = :heap
end

def stable_three(node)
  node.storage = :heap
  node.provenance = :heap
end

def misses_provenance(node)
  node.storage = :heap
end

File: gems/decomplex/examples/php/co-update.php
<?php
function stable_one($node) {
  $node->storage = HEAP;
  $node->provenance = HEAP;
}

function stable_two($node) {
  $node->storage = HEAP;
  $node->provenance = HEAP;
}

function stable_three($node) {
  $node->storage = HEAP;
  $node->provenance = HEAP;
}

function misses_provenance($node) {
  $node->storage = HEAP;
}
